src/components/Reports/ReportsPageRanking/ReportsPageRanking.js

```diff
@@ -1,108 +1,103 @@
-    this.datahubClient = datahubClient;
+const ReportsPageRanking = ({ domainName, dateFrom, dependencies: { datahubClient } }) => {
+  const [state, setState] = useState({ loading: true });
 
-    this.state = {
-      pages: null,
+  useEffect(() => {
+    const fetchData = async () => {
+      const result = await datahubClient.getPagesRankingByPeriod({
+        domainName: domainName,
+        dateFrom: dateFrom,
+      });
+      if (!result.success) {
+        setState({ loading: false });
+      } else {
+        setState({
+          loading: false,
+          pages: result.value,
+        });
+      }
     };
 
-    this.fetchPagesRankingByPeriod = this.fetchPagesRankingByPeriod.bind(this);
-  }
+    fetchData();
+  }, [datahubClient, domainName, dateFrom]);
 
-  async fetchPagesRankingByPeriod(domainName, dateFrom) {
-    this.asyncRequest = this.datahubClient.getPagesRankingByPeriod({
-      domainName: domainName,
-      dateFrom: dateFrom,
-    });
-    const pages = await this.asyncRequest;
-    this.asyncRequest = null;
-    this.setState({
-      pages: pages.value,
-    });
-  }
+  return (
+    <div className="wrapper-reports-box">
+      {state.loading ? (
+        <Loading />
+      ) : (
+        <div className="reports-box">
+          <small className="title-reports-box">
+            <FormattedMessage id="reports_pageranking.top_pages" />
+          </small>
 
-  componentWillUnmount() {
-    // TODO: abort request or at least side effects after finish
-    this.asyncRequest = null;
-  }
-
-  componentDidMount() {
-    this.fetchPagesRankingByPeriod(this.props.domainName, this.props.dateFrom);
-  }
-
-  static getDerivedStateFromProps(props, state) {
-    if (
-      props.domainName !== state.prevDomainName ||
-      props.dateFrom !== state.prevDateFrom ||
-      props.dateTo !== state.prevDateTo
-    ) {
-      return {
-        pages: null,
-        prevDomainName: props.domainName,
-        prevDateFrom: props.dateFrom,
-        prevDateTo: props.dateTo,
-      };
-    }
-
-    // No state update necessary
-    return null;
-  }
-
-  componentDidUpdate(prevProps, prevState) {
-    if (this.state.pages === null) {
-      this.fetchPagesRankingByPeriod(this.props.domainName, this.props.dateFrom);
-    }
-  }
-
-  render() {
-    const {
-      state: { pages },
-    } = this;
-
-    return (
-      <>
-        <div className="wrapper-reports-box">
-          {pages === null ? (
-            <Loading />
-          ) : (
-            <div className="reports-box">
-              <small className="title-reports-box">
-                <FormattedMessage id="reports_pageranking.top_pages" />
-              </small>
-
-              {pages.map((item, index) => (
-                <div key={index} className="page-ranking--item">
-                  <p className="text-ranking">
+          {state.pages ? (
+            state.pages.map((item, index) => (
+              <StyledPageRankingItem key={index}>
+                <div>
```

Comment:
Is this `<div>` necessary?

Reply:
We have 2 columns, this `<div>` is one of that